[TiDB] Add a new mode `prefer-leader` for `replica_read_mode`. (#12337)
pingcap/docs: follower-read.md

- When the value of `tidb_replica_read` is set to `leader` or an empty string, TiDB maintains its default behavior and sends all read operations to the leader replica to perform.
- When the value of `tidb_replica_read` is set to `follower`, TiDB selects a follower replica of the Region to perform all read operations.
- When the value of `tidb_replica_read` is set to `leader-and-follower`, TiDB can select any replicas to perform read operations. In this mode, read requests are load balanced between the leader and follower.

> - When the value of `tidb_replica_read` is set to `prefer-leader`, TiDB selects the leader replica as priority to perform read operations. And if the leader replica is obviously slow on processing such as by Disk / Net performance jitter, TiDB will select other available follower replicas to perform.

Reviewer: ```suggestion
- When the value of `tidb_replica_read` is set to `prefer-leader`, TiDB prefers to select the leader replica to perform read operations. If the leader replica is obviously slow in processing read operations (such as caused by disk or network performance jitter), TiDB will select other available follower replicas to perform read operations.
```
